nominal/core/dataset_file.py

```diff
@@ -241,6 +251,44 @@ def download_original_files(
                 results.append(destination)
             return results
 
+    def poll_until_ingestion_completed(self, interval: datetime.timedelta = datetime.timedelta(seconds=1)) -> Self:
+        """Block until dataset file ingestion has completed
+
+        This method polls Nominal for ingest status after uploading a file to a dataset on an interval.
+
+        """
+        while True:
+            self.refresh()
+            if self.ingest_status is IngestStatus.SUCCESS:
+                break
+            elif self.ingest_status is IngestStatus.IN_PROGRESS:
+                pass
+            elif self.ingest_status is IngestStatus.FAILED:
+                # Get error message to display to user
+                raw_file = self._clients.catalog.get_dataset_file(self._clients.auth_header, self.dataset_rid, self.id)
+                file_error = raw_file.ingest_status.error
+                if file_error is None:
```

Comment:
This is racy, I think we could solve with a `_refresh(ingest_status)` function so that we can use the same response to refresh?

This is different from the datasource poll method, which doesn't refresh on each iteration. This seems more correct though, would you be down to update that?